src/boot/lib/mexpr.ml

```diff
@@ -989,6 +1058,208 @@ let delta eval env fi c v =
       TmSeq (fi, Mseq.Helpers.of_list binds)
   | CmapBindings, _ ->
       fail_constapp fi
+  (* MCore intrinsics: Tensors *)
+  | CTensor _, _ ->
+      fail_constapp fi
+  | CtensorCreate None, TmSeq (_, seq) ->
+      let shape = tm_seq2int_array fi seq in
+      TmConst (fi, CtensorCreate (Some shape))
+  | CtensorCreate (Some shape), (TmClos (clfi, _, _, _, _) as cl) ->
```

Comment:
Matching on a closure here means that e.g. `tensorCreate [] (addi 1)` is not valid, but maybe that's OK.

Reply:
That is not intended, thx! Even though `addi 1` has the wrong type.

Reply:
Right, I realised I misunderstood the type signature of `tensorCreate`. I believe a correct example is `tensorCreate [] length`